src/frequenz/sdk/model/_manager.py

```diff
@@ -0,0 +1,273 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Load, update, monitor and retrieve machine learning models."""
+
+import abc
+import asyncio
+import logging
+import os
+import pickle
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Generic, TypeVar
+
+from frequenz.channels.file_watcher import EventType, FileWatcher
+
+from frequenz.sdk._internal._asyncio import cancel_and_await
+
+_logger = logging.getLogger(__name__)
+
+T = TypeVar("T")
+"""The model type."""
+
+
+class ModelRepository(abc.ABC):
```

Comment:
Can't we parametrize keys in this class with `T` too? I think we should avoid using `Any` as it completely breaks the type system.

```suggestion
class ModelRepository(abc.ABC, Generic[T]):
```